### PR TITLE
5448 implementing cookie information

### PR DESCRIPTION
**Pull request by arniloftsson**
#### Link to issue

https://platform.dandigbib.org/issues/5447 and https://platform.dandigbib.org/issues/5448  

#### Description

This PR removes the existing eu_cookie complience module and its intergrations and implements the new cookieinformation service.

Regarding removing the eu cookie complience module. There are a number of update hooks regarding the module which now are obsolete. The best practise in the existing code seems to be to remove the code leaving a comment. That is the approach I have taken here.

In implementing the new cookieinformation I use the module cookieinformation . 

There is a issue of handling of third party cookies. There are various ways of doing that but I choose letting cookieinformation do most of the work which achieved by altering slightly in the iframe code which is used to integrate third party content. Specifically I altered the template the youtube and vimeo media modules use to output an iframe. And I added a hook for oembed requests to alter iframes which come from ekurser. There are other ways of doing this. Custom js code etc. But this seems to be the simplest approach.

A big question mark in all this is the interplay between cookieinformation and the Mapp service. The nocookie solution provided in Mapp relays on the old cookie module to delete cookies when the user opts out. Therefore I implemented a js function to remove thoose cookies. But there is no much documentation about the Mapp nocookie solution.

#### Screenshot of the result

image

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: f85d367753e9034a5639975587b3a78e4ca3969e
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: f85d367753e9034a5639975587b3a78e4ca3969e
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

File: modules/ding_webtrekk/ding_webtrekk.js [hunk 1]
```diff
@@ -237,20 +237,30 @@
...
-    // there's no method for that, we check for the value explicitly.
-    if ((method === 'opt_in' && !Drupal.eu_cookie_compliance.hasAgreed())
-        || (method === 'opt_out' && Drupal.eu_cookie_compliance.getCurrentStatus() === 0)) {
-      noCookieTracking();
+      const noCookieTracking = function (name) {
+        var cookiesToDelete = Drupal.settings.dingWebtrekk.cookiesToRemove;
+        cookiesToDelete.forEach(cookie => {
+          // The version of jquery.cookie which comes with this version of Drupal does not have
+          // removeCookie function so we use this method instead.
+          $.cookie(cookie, null, { path: '/', domain: window.location.hostname });
```
Comment (kasperg) [inline modules/ding_webtrekk/ding_webtrekk.js:247]: :+1: Appreciate the commentary.

File: modules/ding_webtrekk/js/jquery.cookie-1.4.1.min.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
..."],a):"object"==typeof exports?a(require("jquery")):a(jQuery)}(function(a){function b(a){return h.raw?a:encodeURIComponent(a)}function c(a){return h.raw?a:decodeURIComponent(a)}function d(a){return b(h.json?JSON.stringify(a):String(a))}function e(a){0===a.indexOf('"')&&(a=a.slice(1,-1).replace(/\\"/g,'"').replace(/\\\\/g,"\\"));try{return a=decodeURIComponent(a.replace(g," ")),h.json?JSON.parse(a):a}catch(b){}}function f(b,c){var d=h.raw?b:e(b);return a.isFunction(c)?c(d):d}var g=/\+/g,h=a.cookie=function(e,g,i){if(void 0!==g&&!a.isFunction(g)){if(i=a.extend({},h.defaults,i),"number"==typeof i.expires){var j=i.expires,k=i.expires=new Date;k.setTime(+k+864e5*j)}return document.cookie=[b(e),"=",d(g),i.expires?"; expires="+i.expires.toUTCString():"",i.path?"; path="+i.path:"",i.domain?"; domain="+i.domain:"",i.secure?"; secure":""].join("")}for(var l=e?void 0:{},m=document.cookie?document.cookie.split("; "):[],n=0,o=m.length;o>n;n++){var p=m[n].split("="),q=c(p.shift()),r=p.join("=");if(e&&e===q){l=f(r,g);break}e||void 0===(r=f(r))||(l[q]=r)}return l};h.defaults={},a.removeCookie=function(b,c){return void 0===a.cookie(b)?!1:(a.cookie(b,"",a.extend({},c,{expires:-1})),!a.cookie(b))}});
```
Comment (kasperg) [inline modules/ding_webtrekk/js/jquery.cookie-1.4.1.min.js:2]: Please remove this jQuery cookie version. It does not seem to be used anywhere. Elsewhere the code refers to the version that ships with Drupal.

**Review by kasperg — CHANGES_REQUESTED** [review]
:+1: This seems like an excellent implementation to me.

As I see it the primary things that need to be changed is the removal of the additional version of jQuery.cookie.

Please also look into the code style comments by Scrutinizer.

File: modules/ding_webtrekk/js/jquery.cookie-1.4.1.min.js (hunk 2 above)
Comment (arniloftsson) [reply modules/ding_webtrekk/js/jquery.cookie-1.4.1.min.js]: Removed.

**Comment by arniloftsson** [comment]
There remains 1 Scrutinizer error which from code not changed in this PR.

**Review by kasperg — APPROVED** [review]
:+1: LGTM

**Pull request merged**